### PR TITLE
Fix payer ID missing errors

### PR DESCRIPTION
**Pull request by mnelson4**
## Problem this Pull Request solves

Fixes payments failing when the smart buttons got initialize twice (because of payment method switching) which would result in "payer ID missing" errors.
The problem was that, after selecting smart buttons then switching to a different payment method, the jQuery objects got invalidated because the inputs they were for were removed temporarily from the page. So the fix was to reinitialize those jQuery objects every time we re-select smart buttons. (Except the jQuery object for the payment div, which is the area where the smart buttons were placed, because it's *not* removed when switching payments, and PayPal's JS doesn't work if that gets reinitialized).

This is an alternative fix to https://github.com/eventespresso/eea-paypal-smart-buttons/pull/32, except doesn't unnecessarily re-fetch the payment details when we already had them.

## How has this been tested

* [x] Pay with paypal smart buttons normally
* [x] Register again, but switch payment methods a few times. It should still work (it wouldn't on master)
* [x] Register again, select smart buttons then refresh the page, and pay. It should work as normla.

## Checklist

* [ ] I have read the documentation relating to systems affected by this pull request, see https://github.com/eventespresso/event-espresso-core/tree/master/docs
* [ ] User input is adequately validated and sanitized
* [ ] all publicly displayed strings are internationalized (usually using `esc_html__()`, see https://codex.wordpress.org/I18n_for_WordPress_Developers)
* [ ] My code is tested.
* [ ] My code follows the Event Espresso code style.
* [ ] My code has proper inline documentation.

### CODE REVIEW

**Comment by Pebblo** [comment]
To me, this all looks A-OK, but JS is not my forte so I'm not confident reviewing it.

Assigning this to Darren.

File: assets/src/ee-paypal-smart-buttons.js [hunk 1]
```diff
@@ -125,8 +125,12 @@ function EegPayPalSmartButtons( instanceVars, translations ) {
 	 * Initializes jQuery selected objects so we don't need to query for anything afterwards
 	 */
 	this.initializeObjects = function() {
+		// Only initialize the paymentDiv once. It's not removed when branch switching, and PayPal's JS malfunctions
```
[the remark this answers is not in the document]
Comment (mnelson4) [reply assets/src/ee-paypal-smart-buttons.js]: ah thanks!

**Comment by Pebblo** [comment]
Just to note, I've tested this as is and it prevents the payerID issue, I can swap, change, select, reselect, refresh etc the payment options and it now always works as expected.

**Comment by mnelson4** [comment]
Ok so this has been tested and reviewed, so merging.

**Pull request merged**